packages/mdc-ripple/_mixins.scss

```diff
@@ -141,21 +199,57 @@ $mdc-ripple-common-styles-emitted_: false !default;
   }
 
   // Foreground ripple styles
+
   &.mdc-ripple-upgraded::after {
-    top: 0;
-    left: 0;
     width: var(--mdc-ripple-fg-size, $radius);
     height: var(--mdc-ripple-fg-size, $radius);
-    transform: scale(0);
-    transform-origin: center center;
   }
+}
 
-  &.mdc-ripple-upgraded--unbounded::after {
-    top: var(--mdc-ripple-top, 0);
-    left: var(--mdc-ripple-left, 0);
+//
+// Legacy
+//
+
+@mixin mdc-ripple-color($color: black, $opacity: .06) {
+  // Opacity styles are here (rather than in mdc-ripple-surface) to ensure that opacity is re-initialized for
+  // cases where this mixin is used to override another inherited use of itself,
+  // without needing to re-include mdc-ripple-surface.
+  &::before,
+  &::after {
+    @include mdc-ripple-color_($color, $opacity);
+
+    opacity: 0;
+  }
+
+  // Note: when :active is applied, :focus is already applied, which will effectively double the effect.
+  &:not(.mdc-ripple-upgraded) {
+    &:hover::before,
+    &:focus::before,
+    &:active::after {
+      transition-duration: 85ms;
+      opacity: .6;
+    }
+  }
+
+  &.mdc-ripple-upgraded--background-focused::before {
+    opacity: .99999;
```

Comment:
Should we add a comment explaining why we're using `.99999` instead of `1`? Or do we just assume that most front end devs will know it's related to GPU acceleration something something?

Reply:
This code is identical to what existed previously. This is in the legacy mixin that will be going away in a couple of weeks hopefully :)